test/config/mixed-cjs-esm.test.ts

```diff
@@ -0,0 +1,53 @@
+import * as path from 'path'
+
+import {Config} from '../../src/config'
+
+import {expect, fancy} from './test'
+
+const root = path.resolve(__dirname, 'fixtures/mixed-cjs-esm')
+const p = (p: string) => path.join(root, p)
+
+const withConfig = fancy
+.add('config', () => Config.load(root))
+
+describe('mixed-cjs-esm', () => {
+  withConfig
+  .it('has commandsDir', ({config}) => {
+    expect(config.plugins[0]).to.deep.include({
+      commandsDir: p('src/commands'),
+    })
+  })
+
+  withConfig
+  .stdout()
+  .it('runs mixed-cjs-esm command and prerun & postrun hooks', async ctx => {
+    await ctx.config.runCommand('foo:bar:baz')
+    expect(ctx.stdout).to.equal('running mixed-cjs-esm prerun hook\nit works!\nrunning mixed-cjs-esm postrun hook\n')
+  })
+
+  withConfig
+  .stdout()
+  .it('runs faulty command, only prerun hook triggers', async ctx => {
+    try {
+      await ctx.config.runCommand('foo:bar:fail')
+    } catch {
+      console.log('caught error')
+    }
+    expect(ctx.stdout).to.equal('running mixed-cjs-esm prerun hook\nit fails!\ncaught error\n')
+  })
+
+  withConfig
+  .stdout()
+  .it('runs mixed-cjs-esm command, postrun hook captures command result', async ctx => {
+    await ctx.config.runCommand('foo:bar:test-result')
+    expect(ctx.stdout).to.equal('running mixed-cjs-esm prerun hook\nit works!\nrunning mixed-cjs-esm postrun hook\nreturned success!\n')
+  })
+
+  withConfig
+  .stdout()
+  .it('runs init hook', async ctx => {
+    // to-do: fix union types
```

Comment:
It isn't clear to me what this todo needs to fix

Reply:
These "to-do" statement are lifted directly from `test/config/typescript.test.ts` which the new ESM and mixed ESM / CJS tests used as the example runtime test except for changing around the type of source being tested. 

However there does seem to be a reason for the comment being there; the line:
`await (ctx.config.runHook as any)` will cause TSC to post an error without the `as any`. 

![Screen Shot 2021-05-25 at 11 14 30 PM](https://user-images.githubusercontent.com/311473/119612087-569dad80-bdb0-11eb-8c5b-2d908a1996fa.png)

If you do track down the method signature in `/src/interfaces` there appears to be a union type. 

Looks like @RasPhilCo can comment on it from looking at git history.